sdk/appcenter-storage/src/main/java/com/microsoft/appcenter/storage/TokenManager.java

```diff
@@ -0,0 +1,59 @@
+package com.microsoft.appcenter.storage;
+
+import com.microsoft.appcenter.utils.storage.SharedPreferencesManager;
+import com.google.gson.Gson;
+import com.microsoft.appcenter.storage.models.TokenResult;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Date;
+import java.util.List;
+
+public class TokenManager {
+
+    private final Gson gson;
+
+    private static TokenManager tInstance;
+
+    private TokenManager() {
+        gson = new Gson();
+    }
+
+    public static synchronized TokenManager getInstance() {
+        if (tInstance == null) {
+            tInstance = new TokenManager();
+        }
+        return tInstance;
+    }
+
+    public List<String> ListPartitionNames() {
+        String partitionNameStrings = SharedPreferencesManager.getString(Constants.PARTITION_NAMES);
+        return partitionNameStrings == null ? new ArrayList<String>() : new ArrayList<>(Arrays.asList(gson.fromJson(partitionNameStrings, String[].class)));
+    }
+
+    public TokenResult cachedToken(String partitionName) {
+        TokenResult token = gson.fromJson(SharedPreferencesManager.getString(partitionName), TokenResult.class);
+        if (token != null){
+            long now = new Date().getTime();
+
+            // the token is considered expired

```

Comment:
Capitalize the first character.
Period at the end of the sentence.
We don't use `//`. Use `/* */` even for single line comment.